src/main/kotlin/com/exactpro/th2/check1/rule/RuleFactory.kt

```diff
@@ -0,0 +1,198 @@
+                        "Either old filter or root filter must be set"
+                    }
+                    val filter: RootMessageFilter = if (request.hasRootFilter()) {
+                        request.rootFilter
+                    } else {
+                        request.filter.toRootMessageFilter()
+                    }
+                    val direction = directionOrDefault(request.direction)
+
+                    CheckRuleTask(
+                            request.description,
+                            Instant.now(),
+                            SessionKey(sessionAlias, direction),
+                            request.timeout,
+                            maxEventBatchContentSize,
+                            filter,
+                            request.parentEventId,
+                            streamObservable,
+                            eventBatchRouter
+                    )
+                }
+                onErrorEvent(request.parentEventId) {
+                    Event.start()
+                            .name("Check rule cannot be created")
+                            .type("checkRuleCreation")
+                }
+            }
+
+    fun createSequenceCheckRule(request: CheckSequenceRuleRequest): SequenceCheckRuleTask =
+            ruleCreation(request) {
+                checkAndCreateRule { request ->
+                    check(request.hasParentEventId()) { "Parent event id can't be null" }
+                    check(request.connectivityId.sessionAlias.isNotEmpty()) { "Session alias cannot be empty" }
+                    val sessionAlias: String = request.connectivityId.sessionAlias
+                    val direction = directionOrDefault(request.direction)
+
+                    check((request.messageFiltersList.isEmpty() && request.rootMessageFiltersList.isNotEmpty())
+                            || (request.messageFiltersList.isNotEmpty() && request.rootMessageFiltersList.isEmpty())) {
+                        "Either messageFilters or rootMessageFilters must be set but not both"
+                    }
+
+                    val protoMessageFilters: List<RootMessageFilter> = request.rootMessageFiltersList.ifEmpty {
+                        request.messageFiltersList.map { it.toRootMessageFilter() }
+                    }
+
+                    SequenceCheckRuleTask(
+                            request.description,
+                            Instant.now(),
+                            SessionKey(sessionAlias, direction),
+                            request.timeout,
+                            maxEventBatchContentSize,
+                            request.preFilter,
+                            protoMessageFilters,
+                            request.checkOrder,
+                            request.parentEventId,
+                            streamObservable,
+                            eventBatchRouter
+                    )
+                }
+                onErrorEvent(request.parentEventId) {
```

Comment:
What is the point of passing `request.parentEventId` in this method? You don't use other values. You don't use that ID in this method (only store it into a field). Maybe we should pass that value into `ruleCreation` mathod?